scripts/initramfs/init.x86

```diff
@@ -307,6 +307,10 @@ fi
 print_msg "Checking for a volumio rootfs update on a USB device"
 mkdir /mnt/usb
 for devlink in /dev/sd[a-z]; do
+  if [ "${devlink}" == "/dev/sd[a-z]" ]; then
```

Comment:
Anything against just adding a simple `[ -e ${devlink}]`  instead? That seems more useful than doing a string comparison. 

You could combine it with the other tests..
```bash
  if [ $(lsblk ${devlink} --list -no tran) == usb ] && \
     [ ${BOOT_DEVICE} != ${devlink} ]; then
```

Reply:
```diff
diff --git a/scripts/initramfs/init.x86 b/scripts/initramfs/init.x86
index e09b259..a2c1f29 100755
--- a/scripts/initramfs/init.x86
+++ b/scripts/initramfs/init.x86
@@ -307,13 +307,9 @@ fi
 print_msg "Checking for a volumio rootfs update on a USB device"
 mkdir /mnt/usb
 for devlink in /dev/sd[a-z]; do
-  if [ "${devlink}" == "/dev/sd[a-z]" ]; then
-  # no usb devices present
-    break
-  fi
   # do the first partition of a usb device, must be removable and not the boot device!
-  if [ $(lsblk ${devlink} --list -no tran) == usb ] && \
-     [ ${BOOT_DEVICE} != ${devlink} ]; then
+  if [ -e "${devlink}" ] && [ $(lsblk ${devlink} --list -no tran) == usb ] &&
+    [ ${BOOT_DEVICE} != ${devlink} ]; then
     mount -t vfat ${devlink}1 /mnt/usb >/dev/null 2>&1
     print_msg "Checking ${devlink}1 for a volumio rootfs update"
     if [ -e /mnt/usb/*.fir ]; then
@@ -336,13 +332,9 @@ done
 
 print_msg "Checking for factory reset on a USB device"
 for devlink in /dev/sd[a-z]; do
-  if [ "${devlink}" == "/dev/sd[a-z]" ]; then
-  # no usb devices present
-    break
-  fi
   # do the first partition of a usb device, must be removable and not the boot device!
-  if [ $(lsblk ${devlink} --list -no tran) == usb ] && \
-     [ ${BOOT_DEVICE} != ${devlink} ]; then
+  if [ -e "${devlink}" ] && [ $(lsblk ${devlink} --list -no tran) == usb ] &&
+    [ ${BOOT_DEVICE} != ${devlink} ]; then
     mount -t vfat ${devlink}1 /mnt/usb >/dev/null 2>&1
     print_msg "Checking ${devlink}1 for a factory reset"
     if [ -e /mnt/usb/factory_reset ]; then
```

Reply:
yes, fully agree. Appreciate the suggestions, still a lot to learn.